src/akamai/js/akamaiHD_flash.js

```diff
@@ -201,8 +205,31 @@ require("../../../html5-common/js/utils/constants.js");
      * @returns {boolean} True or false indicating success
      */
     this.setVideoUrl = function(url, encoding) {
-      /* Should be set to false if stream URL changes */
-      loaded = false;
+      var urlChanged = false;
+      newController=this.controller;
+
+      if (currentUrl.replace(/[\?&]_=[^&]+$/,'') != url)
+      {
+        currentUrl = url || "";
+
+        // bust the chrome caching bug
+        if (currentUrl.length > 0) 
+        {
+          currentUrl = currentUrl + (/\?/.test(currentUrl) ? "&" : "?") + "_=" + getRandomString();
+        }
+        urlChanged = true;
+        loaded = false;
+        url = "setVideoUrl("+currentUrl+")";
+      }
+      if (_.isEmpty(currentUrl)) 
+      {
+        this.controller.notify(this.controller.EVENTS.ERROR, { errorcode: 0 }); 
```

Comment:
Actually we might support setting empty string as url for ads as a method of unloading the current stream.  It might be good to reconsider this condition